ColabTool/Shared/PostDetails/index.jsx

```diff
@@ -80,94 +82,93 @@ const renderControls = ({
       <Text size={'small'}> Deleting... </Text>
     );
   }
+
+  if (manager && isWorking && !isDeleting) {
+    return (
+      <Text size={'small'}> Approving... </Text>
+    );
+  }
+
   return (
     <div>
-      {renderEdit({
-        onEditClick,
-      })}
       {renderDelete({
         isConfirmingDelete,
         onCancelConfirmClick,
         onDeleteConfirmClick,
         onDeleteClick,
       })}
+      {renderEdit({
+        onEditClick,
+      })}
+      {manager && renderApproval({
+        onApproveClick,
+      })}
     </div>
   );
 };
 
-const renderPostTypeIcon = (postType) => {
-  switch (postType) {
-    case 'image':
-      return (<ImageIcon size={'small'} />);
-    case 'retweet':
-      return (<RetweetIcon size={'small'} />);
-    default:
-      return (<LinkIcon size={'small'} />);
-  }
-};
-
 /* eslint-enable react/prop-types */
 
 const PostDetails = ({
   isDeleting,
   isConfirmingDelete,
+  isWorking,
+  manager,
+  onApproveClick,
   onCancelConfirmClick,
   onDeleteClick,
   onDeleteConfirmClick,
   onEditClick,
-  profile,
-  postType,
+  draftDetails,
 }) =>
   <div className={style['post-details']}>
+    <div className={style['post-action-details']}>
+      <span className={style['post-icon']}>
```

Comment:
I think we might be able to omit this `span` here so that any vertical alignment can be applied directly on the `Icon`. Right now I _think_ this span is something like 16x22 with the 16x16 icon sitting within it. This span might be 16x22 because of its sibling element with "This post is scheduled..." being naturally 22px in height. If we omitted this span, we could then apply the `margin-right` property directly on the `svg`, so something like:

```css
.post-details svg {
  margin-right: 0.5rem;
}
```